extensions/amp-accordion/1.0/accordion.js

```diff
@@ -88,19 +92,31 @@ function AccordionWithRef(
           expandSingleSection
         );
       });
-      return () => setExpandedMap((expandedMap) => omit(expandedMap, id));
+      eventMap.current[id] = {onExpand, onCollapse};
+      return () => {
+        setExpandedMap((expandedMap) => omit(expandedMap, id));
+        omit(/** @type {!Object} */ (eventMap.current), id);
+      };
     },
     [expandSingleSection]
   );
 
   const toggleExpanded = useCallback(
     (id) => {
+      const newValue = !expandedMap[id];
       setExpandedMap((expandedMap) => {
-        const newValue = !expandedMap[id];
         return setExpanded(id, newValue, expandedMap, expandSingleSection);
       });
+      const onExpand = eventMap.current[id] && eventMap.current[id].onExpand;
+      const onCollapse =
+        eventMap.current[id] && eventMap.current[id].onCollapse;
+      if (newValue && onExpand) {
+        onExpand();
+      } else if (onCollapse) {
```

Comment:
This is a wrong `if`. Per this code, the `onCollapse` will be called when either is true: `newValue !== true` (expected), and `!onExpand` (definitely not expected). I.e. `newValue === true && !onExpand` should not trigger `onCollapse`.

Reply:
Also consider that we only really need one callback, e.g. `onExpandStateChanged(boolean)`

Reply:
Thanks for the catching this.  I went with the single `onExpandStateChange(boolean)` function.